frontend/app/routes/_public+/apply+/$id+/personal-information.tsx

```diff
@@ -31,50 +36,234 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
+    .object({
+      phoneNumber: z
+        .string()
+        .refine((val) => !val || (val && isValidPhoneNumber(val, 'CA')), { message: 'invalid-phone' })
+        .optional(),
+      phoneNumberAlt: z
+        .string()
+        .refine((val) => !val || (val && isValidPhoneNumber(val, 'CA')), { message: 'invalid-phone' })
+        .optional(),
+      mailingAddress: z
+        .string()
+        .min(1, { message: 'empty-field' })
+        .transform((val) => val.trim()),
+      mailingApartment: z.string().trim().optional(),
+      mailingCountry: z
+        .string()
+        .min(1, { message: 'empty-field' })
+        .transform((val) => val.trim()),
+      mailingProvince: z.string().min(1, { message: 'empty-field' }).optional(),
+      mailingCity: z
+        .string()
+        .min(1, { message: 'empty-field' })
+        .transform((val) => val.trim()),
+      mailingPostalCode: z.string().min(1, { message: 'empty-field' }).trim(),
+      copyMailingAddress: z.string().optional(),
+      homeAddress: z.string().optional(),
+      homeApartment: z.string().optional(),
+      homeCountry: z.string().optional(),
+      homeProvince: z.string().optional(),
+      homeCity: z.string().optional(),
+      homePostalCode: z.string().optional(),
+    })
+    .superRefine((val, ctx) => {
+      if (!isValidPostalCode(val.mailingCountry, val.mailingPostalCode)) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: val.mailingCountry === 'CAN' ? 'invalid-postal-code' : 'invalid-zip-code',
+          path: ['mailingPostalCode'],
+        });
+      }
+    });
+
   const formData = Object.fromEntries(await request.formData());
-  const parsedDataResult = applyFlow.personalInfoStateSchema.safeParse(formData);
+  const { copyMailingAddress } = formData as Partial<z.infer<typeof copyMailingAddressSchema>>;
+
+  let parsedDataResult;
+  //only parse mailing address since home address will be copied
+  if (copyMailingAddress === 'on') {
+    parsedDataResult = mailingAddressOnlyFormSchema.safeParse(formData);
+  } else {
+    parsedDataResult = personalInformationFormSchema.safeParse(formData);
+  }
 
   if (!parsedDataResult.success) {
     return json({
       errors: parsedDataResult.error.format(),
-      formData: formData as Partial<z.infer<typeof applyFlow.personalInfoStateSchema>>,
+      formData: formData as Partial<z.infer<typeof personalInformationFormSchema>>,
     });
   }
 
+  let saveData = parsedDataResult.data;
+  if (parsedDataResult.data.copyMailingAddress === 'on') {
+    saveData = {
+      ...parsedDataResult.data,
+      homeAddress: parsedDataResult.data.mailingAddress,
+      homeApartment: parsedDataResult.data.mailingApartment,
+      homeCountry: parsedDataResult.data.mailingCountry,
+      homeProvince: parsedDataResult.data.mailingProvince,
+      homeCity: parsedDataResult.data.mailingCity,
+      homePostalCode: parsedDataResult.data.mailingPostalCode,
+    };
+  }
```

Comment:
**Nitpick:** We can change this to a `const` instead as we know this value will never be re-assigned. 

```suggestion
  const savedData =
    parsedDataResult.data.copyMailingAddress === 'on'
      ? {
          ...parsedDataResult.data,
          homeAddress: parsedDataResult.data.mailingAddress,
          homeApartment: parsedDataResult.data.mailingApartment,
          homeCountry: parsedDataResult.data.mailingCountry,
          homeProvince: parsedDataResult.data.mailingProvince,
          homeCity: parsedDataResult.data.mailingCity,
          homePostalCode: parsedDataResult.data.mailingPostalCode,
        }
      : parsedDataResult.data;
```

Reply:
fixed